### PR TITLE
Log which filters have been used

### PR DESCRIPTION
**Pull request by renaudallard**
This is useful to know which filters have been used to filter out data if you enabled multiple filters.

### CODE REVIEW

File: src/org/traccar/FilterHandler.java [hunk 1]
```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
 
     private boolean filter(Position p) {
 
+        String filterType = "";
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: You should use `StringBuilder` instead.

File: src/org/traccar/FilterHandler.java [hunk 2]
```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
...
+
         boolean result = filterInvalid(p) || filterZero(p) || filterDuplicate(p)
                 || filterFuture(p) || filterApproximate(p) || filterStatic(p)
                 || filterDistance(p);
 
+        if (filterInvalid(p)) {
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: You are checking every filter twice. I think it's not a good idea. Filter check should be combined with logging.

File: src/org/traccar/FilterHandler.java [hunk 3]
```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
...
+            filterType = "Invalid ";
+        }
+        if (filterZero(p)) {
+            filterType = (filterType + "Zero ");
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: There is no need for parentheses here. Applies to all other cases as well.

File: src/org/traccar/FilterHandler.java [hunk 4]
```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
...
     private boolean filter(Position p) {
 
+        String filterType = "";
+
         boolean result = filterInvalid(p) || filterZero(p) || filterDuplicate(p)
                 || filterFuture(p) || filterApproximate(p) || filterStatic(p)
                 || filterDistance(p);
 
+        if (filterInvalid(p)) {
+            filterType = "Invalid ";
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: This case shouldn't be different from others. For consistency and easier modifications in future.

File: src/org/traccar/FilterHandler.java [hunk 5]
```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
 
     private boolean filter(Position p) {
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: If you break down filters into separate ifs, please rename the variable to "position".

File: src/org/traccar/FilterHandler.java (hunk 2 above)
Comment (renaudallard) [inline src/org/traccar/FilterHandler.java]: Yes, I don't like it either, but the filterLimit should be taken into account

File: src/org/traccar/FilterHandler.java [hunk 6]
```diff
@@ -150,16 +150,37 @@ private boolean filterLimit(Position position) {
...
+            filterType.append("Approximate ");
+        }
+        if (filterStatic(p)) {
+            filterType.append("Static ");
+        }
+        if (filterDistance(p)) {
+            filterType.append("Distance ");
+        }
+
+        if (!"".equals(filterType.toString()) && !filterLimit(p)) {
```
Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: Please no. This is definitely not the right way.

Comment (renaudallard) [inline src/org/traccar/FilterHandler.java]: But I don't like putting a result = true in every if either

Comment (tananaev) [inline src/org/traccar/FilterHandler.java]: At least it should be `!filterType.isEmpty()`.

**Comment by renaudallard** [comment]
Anything else is not OK for you?

**Comment by tananaev** [comment]
I have some more comments, but I'll fix them myself.

**Pull request merged**